src/main/java/seedu/duke/module/Module.java

```diff
@@ -65,4 +69,42 @@ public void markTaskInList(Task task) {
     public void unmarkTaskInList(Task task) {
         task.setDone(false);
     }
+
+    public ArrayList<Task> getTasksToDelete(UI ui, String message, String commandVerb) {
+        ui.printGetChosenTasksPrompt(message, commandVerb, taskList);
+        return getChosenTasks(ui, taskList);
+    }
+
+    public ArrayList<Task> getTasksToMarkOrUnmark(UI ui, String message, String commandVerb, boolean isDone) {
+        ArrayList<Task> tasksToModify = getFilteredTasks(isDone);
+        ui.printGetChosenTasksPrompt(message, commandVerb, tasksToModify);
+        return getChosenTasks(ui, tasksToModify);
+    }
+
+    public ArrayList<Task> getFilteredTasks(boolean isDone) {
+        ArrayList<Task> filteredTasks = new ArrayList<>();
+        for (Task task : taskList) {
+            if (task.getDone() == isDone) {
+                filteredTasks.add(task);
+            }
+        }
+        return filteredTasks;
+    }
```

Comment:
Perhaps these two methods can combine since the method isn't long, as "getFilteredTasks" can easily be confused with "getChosenTasks" below.